hazelcast/src/main/java/com/hazelcast/client/impl/clientside/ClientSchemaService.java

```diff
@@ -21,24 +21,39 @@
 import com.hazelcast.client.impl.protocol.codec.ClientSendAllSchemasCodec;
 import com.hazelcast.client.impl.protocol.codec.ClientSendSchemaCodec;
 import com.hazelcast.client.impl.spi.impl.ClientInvocation;
+import com.hazelcast.cluster.Member;
 import com.hazelcast.internal.serialization.impl.compact.Schema;
 import com.hazelcast.internal.serialization.impl.compact.SchemaService;
 import com.hazelcast.logging.ILogger;
+import com.hazelcast.spi.properties.HazelcastProperties;
+import com.hazelcast.spi.properties.HazelcastProperty;
 
 import javax.annotation.Nullable;
 import java.util.ArrayList;
 import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
 import java.util.concurrent.ConcurrentHashMap;
 
+import static com.hazelcast.client.properties.ClientProperty.INVOCATION_RETRY_PAUSE_MILLIS;
+
 public class ClientSchemaService implements SchemaService {
 
-    private final Map<Long, Schema> schemas = new ConcurrentHashMap<>();
+    public static final HazelcastProperty MAX_PUT_RETRY_COUNT =
+            new HazelcastProperty("hazelcast.client.schema.max.put.retry.count", 100);
```

Comment:
Instead of the count based retry, we talked about using client invocation timeout as an alternative, do you consider using it at all? And it is user configurable.

Reply:
It is not different from what we have here, but we can do that if you want. I feel like this one is a bit more secure